pkg/cmd/pulumi/config.go

```diff
@@ -996,13 +996,26 @@ func looksLikeSecret(k config.Key, v string) bool {
 		(info.Entropy >= (entropyThreshold/2) && entropyPerChar >= entropyPerCharThreshold)
 }
 
+func getAndSaveSecretsManager(stack backend.Stack, workspaceStack *workspace.ProjectStack) (secrets.Manager, error) {
+	sm, needsSave, err := getStackSecretsManager(stack, workspaceStack)
+	if err != nil {
+		return nil, fmt.Errorf("get stack secrets manager: %w", err)
+	}
+	if needsSave {
+		if err = saveProjectStack(stack, workspaceStack); err != nil {
+			return nil, fmt.Errorf("save stack config: %w", err)
+		}
+	}
+	return sm, nil
+}
+
 // getStackConfiguration loads configuration information for a given stack. If stackConfigFile is non empty,
 // it is uses instead of the default configuration file for the stack
 func getStackConfiguration(
 	ctx context.Context,
 	stack backend.Stack,
 	project *workspace.Project,
-	sm secrets.Manager,
+	fallbackSecretsManager *secrets.Manager,
```

Comment:
secrets.Manager is an interface. It doesn't need to be a pointer.
I would just add a comment next to it to indicate it's optional/may be nil.

```suggestion
	fallbackSecretsManager secrets.Manager, // optional
```